src/coll/mod.rs

```diff
@@ -335,108 +425,124 @@ impl Collection {
-            } else {
-                WriteModel::DeleteOne { filter: model.filter.clone() }
-            }
-        }).collect();
+    fn execute_update_batch(&self,
+                            models: Vec<UpdateModel>,
+                            start_index: i64,
+                            ordered: bool,
+                            result: &mut BulkWriteResult,
+                            exception: &mut BulkWriteException)
+                            -> bool {
+        let original_models = models.iter()
+            .map(|model| {
+                if model.multi {
+                    WriteModel::DeleteMany { filter: model.filter.clone() }
+                } else {
+                    WriteModel::DeleteOne { filter: model.filter.clone() }
+                }
+            })
+            .collect();
 
         match self.bulk_update(models, ordered, None, CommandType::UpdateMany) {
-            Ok(bulk_update_result) =>
+            Ok(bulk_update_result) => {
                 result.process_bulk_update_result(bulk_update_result,
-                                                  original_models, start_index,
-                                                  exception),
-            Err(_) =>  {
+                                                  original_models,
+                                                  start_index,
+                                                  exception)
+            }
+            Err(_) => {
                 exception.add_unproccessed_models(original_models);
                 false
             }
         }
     }
 
-    fn execute_batch(&self, batch: Batch, start_index: i64, ordered: bool,
+    fn execute_batch(&self,
+                     batch: Batch,
+                     start_index: i64,
+                     ordered: bool,
                      result: &mut BulkWriteResult,
-                     exception: &mut BulkWriteException) -> bool {
+                     exception: &mut BulkWriteException)
+                     -> bool {
         match batch {
-            Batch::Insert(docs) =>
-                self.execute_insert_batch(docs, start_index, ordered, result,
-                                          exception),
-            Batch::Delete(models) =>
-                self.execute_delete_batch(models, ordered, result,
-                                          exception),
-            Batch::Update(models) =>
-                self.execute_update_batch(models, start_index, ordered, result,
-                                          exception),
+            Batch::Insert(docs) => {
+                self.execute_insert_batch(docs, start_index, ordered, result, exception)
+            }
+            Batch::Delete(models) => self.execute_delete_batch(models, ordered, result, exception),
+            Batch::Update(models) => {
+                self.execute_update_batch(models, start_index, ordered, result, exception)
+            }
```

Comment:
pretty surprised it handled these cases differently -- seems like they're all the same length! Does it just ensure the first and last are in a block if there are more than two cases?